### PR TITLE
Issue #8019: resolve FileTabCharacter violations in test resources

### LINKED ISSUE #8019: resolve FileTabCharacter violations in test resources
leftover from #4100 , detected at https://github.com/checkstyle/checkstyle/pull/7775#discussion_r402314244

FileTabCharacter is still disabled at `checkstyle_resources_checks.xml`

When activated there bunch violations `File contains tab characters (this is the first instance). [FileTabCharacter]`

Some Inputs are created with tabs by design, so such files should be placed to suppress at `checkstyle_resources_suppressions.xml` but better to have `Tab` in name.

```
src/it/resources/com/google/checkstyle/test/chapter2filebasic/rule231filetab/InputFileTabCharacter.java:8:25: 
src/it/resources/org/checkstyle/suppressionxpathfilter/importorder/SuppressionXpathRegressionImportOrderFive.java:8:1: 
src/it/resources/org/checkstyle/suppressionxpathfilter/importorder/SuppressionXpathRegressionImportOrderFour.java:8:1: 
src/it/resources/org/checkstyle/suppressionxpathfilter/importorder/SuppressionXpathRegressionImportOrderOne.java:7:1: 
src/it/resources/org/checkstyle/suppressionxpathfilter/importorder/SuppressionXpathRegressionImportOrderTwo.java:8:1: 
src/it/resources/org/checkstyle/suppressionxpathfilter/invalidjavadocposition/SuppressionXpathRegressionInvalidJavadocPositionSix.java:4:1: 
src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationLang.java:13:1: 
src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/asttreestringprinter/InputAstTreeStringPrinterComments.java:5:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/asttreestringprinter/InputAstTreeStringPrinterFullOfBlockComments.java:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/asttreestringprinter/InputAstTreeStringPrinterJavadocPosition.java:5:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTabCharacter.txt:2:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/magicnumber/InputMagicNumberAnnotationElement.java:7:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistance.java:5:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/InputInnerTypeLastClass.java:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/commentsindentation/InputCommentsIndentationCommentIsAtTheEndOfBlock.java:214:3: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnonymousClassInMethod.java:19:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationCustomAnnotation.java:100:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationFromGuava.java:48:5: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationInvalidSwitchIndent.java:99:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationLambda3.java:14:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationLambda4.java:13:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationLambda5.java:5:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationMethodCallLineWrap.java:34:5: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationUseTabs.java:19:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidMethodIndent.java:198:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCorrectParagraph.java:82:5: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocPosition.java:14:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocPositionWithSinglelineComments.java:14:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocblocktaglocation/InputJavadocBlockTagLocationIncorrect.java:14:6: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocstyle/InputJavadocStyle.java:322:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameOverridableMethod.java:6:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameType.java:38:5: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/methodname/InputMethodNameExtra.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/linelength/InputLineLengthSimple.java:19:25: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentSimple.java:19:25: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/InputFileTabCharacterSimple.java:19:25: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java:26:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nolinewrap/InputNoLineWrapBad.java:11:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nolinewrap/InputNoLineWrapGood.java:12:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfter.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeDot.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeDotAllowLineBreaks.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/parenpad/InputParenPadLeftRightAndNoSpace.java:161:9: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/parenpad/InputParenPadWhitespace.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorComments.java:13:20: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java:14:8: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/typecastparenpad/InputTypecastParenPadWhitespace.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/typecastparenpad/InputTypecastParenPadWhitespaceAround.java:26:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterAround.java:26:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterDefaultConfig.java:19:25: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterTypeCast.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAround.java:26:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundArrayInitialization.java:18:9: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundEmptyTypesAndCycles.java:19:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundKeywordsAndOperators.java:226:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundSimple.java:19:25: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithnearbycommentfilter/InputSuppressWithNearbyCommentFilter.java:67:21: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/InputSuppressWithPlainTextCommentFilter.sql:2:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/InputSuppressWithPlainTextCommentFilter.xml:7:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/InputSuppressWithPlainTextCommentFilterCustomMessageFormat.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/InputSuppressWithPlainTextCommentFilterSuppressById.java:9:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/InputSuppressWithPlainTextCommentFilterWithCustomOnAndOffComments.java:5:7: 
src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/InputSuppressWithPlainTextCommentFilterWithDefaultCfg.java:5:7: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputJava7Diamond.java:6:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJavaClass1.java:25:14: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/comments/InputComments2.java:12:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/comments/InputFullOfBlockComments.java:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations1.java:7:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations10.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations12.java:15:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations3.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations4.java:9:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations5.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations6.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations7.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations8.java:9:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputAnnotations9.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputDefaultMethods.java:5:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputDefaultMethods2.java:5:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda1.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda10.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda11.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda12.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda13.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda14.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda16.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda2.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda3.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda4.java:9:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda5.java:10:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda6.java:9:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda7.java:9:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda8.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputLambda9.java:8:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java8/InputMethodReferences.java:13:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/InputJavadocWithCrAsNewline.javadoc:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/InputLeadingAsterisksExtended.javadoc:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/htmlTags/InputListWithUnclosedItemInUnclosedParagraph.javadoc:3:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/htmlTags/InputMixedCaseOfHtmlTags.javadoc:7:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/htmlTags/InputUnclosedParagraphFollowedByJavadocTag.javadoc:2:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/main/InputMainGenerateXpathSuppressionsTabWidth.java:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/main/InputMainSuppressionsStringPrinter.java:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/xpath/xpathquerygenerator/InputXpathQueryGeneratorTabWidth.java:4:1: 
src/test/resources/com/puppycrawl/tools/checkstyle/xpathfilegeneratorastfilter/InputXpathFileGeneratorAstFilter.java:6:1: 

```

### PR DESCRIPTION
**Pull request by shashwat70**
Resolves #8019 
Out of the 105 violating files, 58 have been fixed and the rest suppressed. Some suppressions are inevitable as they are required for testing. The others have been suppressed because fixing them leads to build failures. However, these can be fixed with some tweaking and these fixes will follow in subsequent PRs by me, once this is merged.

### CODE REVIEW

**Comment by shashwat70** [comment]
Please review and merge, so that I may start working on the remaining issue. @romani @rnveach

**Comment by shashwat70** [comment]
@romani @rnveach ping

File: config/checkstyle_resources_suppressions.xml [hunk 1]
```diff
@@ -74,4 +74,78 @@
...
+  <suppress checks="FileTabCharacter"
+    files="[\\/]InputFileTabCharacterSimple\.java"/>
+  <suppress checks="FileTabCharacter"
+    files="[\\/]InputIndentationUseTabs\.java"/>
+  <suppress checks="FileTabCharacter"
+    files="[\\/]InputCheckerTabCharacter\.txt"/>
+  <suppress checks="FileTabCharacter"
+    files="src[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]indentation[\\/]indentation[\\/]Input.*\.java"/>
+  <suppress checks="FileTabCharacter"
+    files="src[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]filters[\\/]suppresswithplaintextcommentfilter[\\/]Input.*"/>
```
Comment (rnveach) [inline config/checkstyle_resources_suppressions.xml:103]: Please trim path to just `[\\/]test[\\/]resources[\\/]` and checkname (indentation, suppresswithplaintext...) with the input on the end.

Comment (shashwat70) [reply config/checkstyle_resources_suppressions.xml:107]: That will suppress more files than what is necessary. Moreover, these will be fixed in subsequent PRs if possible. Please refer PR description.

**Comment by shashwat70** [comment]
@rnveach ping

**Comment by romani** [comment]
@shashwat70 , thanks a lot for huge update, sorry for delay to accept it.

**Pull request merged**